File: AspNetMvcUnleashed/0672329980_source/Chapter07/VB/MvcApplication1/MvcApplication1/Controllers/CompanyController.vb
Imports System.Security.Principal

Public Class CompanyController
    Inherits Controller

    Public Function GetSecret(<ModelBinder(GetType(UserModelBinder))> ByVal user As IPrincipal) As String
        If user.Identity.Name = "CEO" Then
            Return "The secret is 42."
        Else
            Return "You are not authorized!"
        End If
    End Function

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter07/CS/MvcApplication1/MvcApplication1/Controllers/CompanyController.cs
using System.Web.Mvc;
using MvcApplication1.CustomModelBinders;
using System.Security.Principal;

namespace MvcApplication1.Controllers
{
    public class CompanyController : Controller
    {

        public string GetSecret([ModelBinder(typeof(UserModelBinder))]IPrincipal user)
        {
            if (user.Identity.Name == "CEO")
                return "The secret is 42.";
            else
                return "You are not authorized!";
        }

    }
}
